packages/tpetra/CMakeLists.txt

```diff
@@ -24,6 +24,173 @@ IF (NOT ${PROJECT_NAME}_ENABLE_CXX11 AND NOT TPL_ENABLE_CUDA)
+      IF (NOT (MPI_BASE_DIR STREQUAL "") AND (IS_DIRECTORY "${MPI_BASE_DIR}/bin"))
+        FIND_PROGRAM (Tpetra_OMPI_INFO_EXECUTABLE "ompi_info" "${MPI_BASE_DIR}/bin" NO_DEFAULT_PATH)
+        IF (Tpetra_OMPI_INFO_EXECUTABLE)
+          SET (Tpetra_FOUND_OMPI_INFO_EXECUTABLE ON)
+        ENDIF ()
+      ENDIF ()
+    ENDIF ()
+
+    IF (Tpetra_FOUND_OMPI_INFO_EXECUTABLE)
+      MESSAGE (STATUS "  - Calling \"ompi_info\" (full path: ${Tpetra_OMPI_INFO_EXECUTABLE}) to find out whether MPI is CUDA aware.")
+
+      # ompi_info produces a LOT of output.  Making CMake capture and
+      # parse all of it could be slow.  Instead, we use "grep" to get
+      # only the line we want.
+      EXECUTE_PROCESS (COMMAND ${Tpetra_OMPI_INFO_EXECUTABLE} --parsable --all
+                       COMMAND "grep" "mpi_built_with_cuda_support:value"
+                       OUTPUT_VARIABLE Tpetra_OMPI_INFO_OUTPUT)
+      IF (DEFINED Tpetra_OMPI_INFO_OUTPUT)
+        # The output to stdout may have an endline, so we have to do a
+        # string search, not just a comparison.  It's possible that
+        # neither "true" nor "false" are found, so we test both.
+        STRING (FIND "${Tpetra_OMPI_INFO_OUTPUT}" "mca:mpi:base:param:mpi_built_with_cuda_support:value:true" Tpetra_OMPI_INFO_OUTPUT_FOUND_TRUE)
+        STRING (FIND "${Tpetra_OMPI_INFO_OUTPUT}" "mca:mpi:base:param:mpi_built_with_cuda_support:value:false" Tpetra_OMPI_INFO_OUTPUT_FOUND_FALSE)
+
+        IF (NOT (Tpetra_OMPI_INFO_OUTPUT_FOUND_TRUE EQUAL -1))
+          MESSAGE (STATUS "  - \"ompi_info\" explicitly claims that your MPI implementation is CUDA aware.")
+          SET (Tpetra_ASSUME_CUDA_AWARE_MPI_DEFAULT ON)
+        ELSEIF (NOT (Tpetra_OMPI_INFO_OUTPUT_FOUND_FALSE EQUAL -1))
+          MESSAGE (STATUS "  - \"ompi_info\" explicitly claims that your MPI implementation is NOT CUDA aware.  You may want to use a different OpenMPI installation that is CUDA aware, or reconfigure and rebuild OpenMPI in order to make it CUDA aware.  For details, please refer to OpenMPI's website: https://www.open-mpi.org/faq/?category=runcuda")
+        ELSE ()
+          MESSAGE (STATUS "  - \"ompi_info\" claims to know nothing about whether your MPI implementation is CUDA aware.  Its output is \"${Tpetra_OMPI_INFO_OUTPUT}\".")
+        ENDIF ()
+      ELSE ()
+        MESSAGE (STATUS "  - While I found the \"ompi_info\" executable, it would not run.  Thus, I will make the sane assumption that your MPI implementation is NOT CUDA aware.")
+      ENDIF ()
+    ELSE ()
+      MESSAGE (STATUS "  - Tpetra did not find the \"ompi_info\" executable.  This may not be bad; for example, if your MPI implementation is not OpenMPI, then you won't have this executable.  Tpetra will conservatively assume that your MPI implementation is NOT CUDA aware.  If you would like to change this, please set this CMake variable Tpetra_ASSUME_CUDA_AWARE_MPI:BOOL=ON explicitly at configure time.")
+    ENDIF ()
+  ENDIF ()
+ENDIF ()
```

Comment:
Could you give the arguments to these ENDIFs? I believe the outermost if statement is quite long